sky/skylet/subprocess_daemon.py

```diff
@@ -28,29 +55,34 @@
     if process is None:
         sys.exit()
 
+    children = []
     if parent_process is not None:
         # Wait for either parent or target process to exit.
         while process.is_running() and parent_process.is_running():
+            try:
+                # process.children() must be called while the target process
+                # is alive, as it will return an empty list if the target
+                # process has already terminated.
+                tmp_children = process.children(recursive=True)
+                if tmp_children:
+                    children = tmp_children
+                    children.append(process)
```

Comment:
Should this line be moved outside the `if` and `while` block, as we should try to terminate the process anyway?

Reply:
@Michaelvll Thanks for the catch. Done.